### DISCUSSION TITLE
【V5】Legend Marker 自定义 string 的时候不生效

### DISCUSSION THREAD

**Discussion by BinghuiXie** [antvis/G2#5317] [Q&A] [answered]
- **G2 Version**:
- **Platform**:
- **Mini Showcase(like screenshots)**:
- **CodePen Link**:

如图所示
![image](https://github.com/antvis/G2/assets/34343589/b0b04a0c-c20c-46f8-92e7-85ccb5234587)
配置 `itemMarker` 为字符串的时候图例 marker 不生效。

根据官方文档，`itemMarker` 只支持 `DisplayObject` 或者对应的函数，但是默认值又给了一个 `circle`
![image](https://github.com/antvis/G2/assets/34343589/ab10fbb7-ed54-426e-b933-3214da55ca68)
如果没有一些内置的 marker，通过 `DisplayObject` 或者函数的形式实现一个不同样式的 marker 是否成本太高？
![image](https://github.com/antvis/G2/assets/34343589/d9c24800-014a-44fb-a786-605f723f059a)
官方 demo 有一个是设置了 string 类型的 itemMarker，也没有展示出来

PS：
看了下代码，问题应该在 `legendCategory.ts` 这个文件里面
![20230716203509_rec_](https://github.com/antvis/G2/assets/34343589/dd2856ea-4fb2-4661-b5d8-ae6f5c5df4f4)

这里在拷贝的时候，`style` 的优先级是最高的，会覆盖 `legendStyle` 的同名属性，而由于自己配置的 `itemMarker` 会被解构在 style 里面；而在前一步给 `legendStyle` 推断 marker shape（`inferItemMarker`） 的时候，会生成同名的 `legendMarker` 函数。
因此 `legendStyle` 里面的会被覆盖

![image](https://github.com/antvis/G2/assets/34343589/5c473744-d1b4-496d-9483-fe5b3497a3e2)

在实际绘制的时候，会直接 `document.createElement('point')`，所以不会生效

**Comment by pearmini** [comment, marked as answer]
最新版本似乎没有问题？

![image](https://github.com/antvis/G2/assets/49330279/aa065ce0-213b-41fc-8a95-b0171107fa5f)
